src/main/java/org/synyx/urlaubsverwaltung/account/AccountFormValidator.java

```diff
@@ -86,46 +88,20 @@ void validatePeriod(AccountForm form, Errors errors) {
         }
     }
 
-    private void validateDateNotNull(LocalDate date, String field, Errors errors) {
-
-        // may be that date field is null because of cast exception, than there is already a field error
-        if (date == null && errors.getFieldErrors(field).isEmpty()) {
-            errors.rejectValue(field, ERROR_MANDATORY_FIELD);
-        }
-    }
-
     void validateAnnualVacation(AccountForm form, Errors errors) {
 
         BigDecimal annualVacationDays = form.getAnnualVacationDays();
-        Settings settings = settingsService.getSettings();
-        AccountSettings accountSettings = settings.getAccountSettings();
-        BigDecimal maxDays = BigDecimal.valueOf(accountSettings.getMaximumAnnualVacationDays());
 
         validateNumberNotNull(annualVacationDays, ATTRIBUTE_ANNUAL_VACATION_DAYS, errors);
 
         if (annualVacationDays != null) {
-            validateNumberOfDays(annualVacationDays, ATTRIBUTE_ANNUAL_VACATION_DAYS, maxDays, errors);
-        }
-    }
-
-    private void validateNumberNotNull(BigDecimal number, String field, Errors errors) {
-
-        // may be that number field is null because of cast exception, than there is already a field error
-        if (number == null && errors.getFieldErrors(field).isEmpty()) {
-            errors.rejectValue(field, ERROR_MANDATORY_FIELD);
-        }
-    }
 
-    private void validateNumberOfDays(BigDecimal days, String field, BigDecimal maximumDays, Errors errors) {
-
-        // is number of days < 0 ?
-        if (days.compareTo(BigDecimal.ZERO) < 0) {
-            errors.rejectValue(field, ERROR_ENTRY);
-        }
+            validateIsInteger(annualVacationDays, ATTRIBUTE_ANNUAL_VACATION_DAYS, errors);
 
-        // is number of days unrealistic?
-        if (days.compareTo(maximumDays) > 0) {
-            errors.rejectValue(field, ERROR_ENTRY);
+            Settings settings = settingsService.getSettings();
```

Comment:
Man könnte einmal die Settings zu begin holen die Parameter dann in die Methoden übergeben. Dann hätte man im max ein Datenbankcall. Was meinst du?